### PR TITLE
Specify che-editors.yaml in devfile v2 format

### PR DESCRIPTION
**Pull request by ericwill**
Signed-off-by: Eric Williams <ericwill@redhat.com>

### What does this PR do?
Converts che-editors.yaml to use devfile v2 format.
Based on: https://github.com/devfile/devworkspace-operator/blob/main/samples/plugins/theia-next.yaml

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Fixes eclipse/che#19034

### How to test this PR?

Build the plugin registry with this change and make sure everything is generated correctly.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by ericwill** [comment]
I wasn't sure exactly of the formatting so this is a first attempt. cc @benoitf can you let me know if I am on the right track

**Comment by benoitf** [comment]
Hello Eric, it seems the format you are using is not following devfile 2 format (you're using a spec field but it's not there in devfile 2.0)
Also there is a vsixinstaller component but I really don't know from where it's coming


IMHO the easiest way to start is the following:

Take generated devfile.yaml

then, we do:
```
version: 1.0.1 (here we bump the definition for plug-in registry, could be 2.0.0 as well)
editors:
  - <copy-paste of the generated devfile.yaml>, for example https://che-plugin-registry-main.surge.sh/v3/plugins/eclipse/che-theia/next/devfile.yaml
  ```
  
add all root attributes like displayName, description, icon, repository, firstPublicationDate to metadata/attributes 
  
something like https://gist.github.com/benoitf/49f9e19929a1d7602951e4306384361f ?

cc @l0rd @davidfestal if you have some other comments

**Comment by ericwill** [comment]
> Hello Eric, it seems the format you are using is not following devfile 2 format (you're using a spec field but it's not there in devfile 2.0)
> Also there is a vsixinstaller component but I really don't know from where it's coming
> 
> IMHO the easiest way to start is the following:
> 
> Take generated devfile.yaml
> 
> then, we do:
> 
> ```
> version: 1.0.1 (here we bump the definition for plug-in registry, could be 2.0.0 as well)
> editors:
>   - <copy-paste of the generated devfile.yaml>, for example https://che-plugin-registry-main.surge.sh/v3/plugins/eclipse/che-theia/next/devfile.yaml
> ```
> 
> add all root attributes like displayName, description, icon, repository, firstPublicationDate to metadata/attributes
> 
> something like https://gist.github.com/benoitf/49f9e19929a1d7602951e4306384361f ?
> 
> cc @l0rd @davidfestal if you have some other comments

Thanks for the pointers, I've adapted my PR. A few questions:
* I used the `id` field from the existing `che-editors.yaml` for the `metadata/name` field
* Instead of setting `version` to `latest`, I set it to the version we have in `che-editors.yaml`
* I used existing images with tags instead of using the generated digests

File: che-editors.yaml [hunk 1]
```diff
@@ -1,373 +1,442 @@
...
+      name: eclipse/che-theia/next
+      displayName: theia-ide
+      description: Eclipse Theia, get the latest release each day.
+      icon: /v3/images/eclipse-che-logo.png
+      attributes:      
+        publisher: eclipse
+        version: next
+        type: Che Editor
+        title: Eclipse Theia development version.
+        category: Editor
```
Comment (benoitf) [inline che-editors.yaml:14]: I think type and category can be removed as inside che-editors we expect to see editors ?

File: che-editors.yaml [hunk 2]
```diff
@@ -1,373 +1,442 @@
...
+        repository: 'https://github.com/eclipse/che-theia'
+        firstPublicationDate: '2019-03-07'
+        latestUpdateDate: '2021-03-18'
```
Comment (benoitf) [inline che-editors.yaml:17]: this one is generated, so it should be committed

File: che-editors.yaml [hunk 3]
```diff
@@ -1,373 +1,442 @@
...
+          endpoints:
+            - name: theia
+              attributes:
+                type: ide
+                cookiesAuthEnabled: true
+                discoverable: false
+              targetPort: 3100
+              exposure: public
+              secure: false
+              protocol: http
```
Comment (benoitf) [inline che-editors.yaml:56]: it's secured:true protocol http for che-server

and secure:false with https protocol for DevWorkspace

so as long as meta.yaml and devfile.yaml are generated correctly I'm fine

File: che-editors.yaml [hunk 4]
```diff
@@ -1,373 +1,442 @@
...
-        ports:
-          - exposedPort: 3333
-        command:
-          - /go/bin/che-machine-exec
-          - '--url'
-          - '127.0.0.1:3333'
-    initContainers:
+        container:
+          image: 'quay.io/eclipse/che-machine-exec:nightly'
+          args:
```
Comment (benoitf) [inline che-editors.yaml:118]: it's turning to command

File: che-editors.yaml [hunk 5]
```diff
@@ -1,373 +1,442 @@
...
+              value: remote-endpoint
+          volumeMounts:
+            - name: remote-endpoint
+              path: /remote-endpoint
+      - name: remote-endpoint
+        volume:
+          ephemeral: true
+  - schemaVersion: 2.1.0
+    metadata:
+      name: eclipse/che-theia/7.27.2
```
Comment (benoitf) [inline che-editors.yaml:138]: if we merge after 7.28.0 we should probably bump to 7.28.0

File: che-editors.yaml [hunk 6]
```diff
@@ -1,373 +1,442 @@
...
-        ports:
-          - exposedPort: 3333
-        command:
-          - /go/bin/che-machine-exec
-          - '--url'
-          - '127.0.0.1:3333'
-    initContainers:
+        container:
+          image: "quay.io/eclipse/che-machine-exec:7.27.2"
+          args:
```
Comment (benoitf) [inline che-editors.yaml:249]: command

File: che-editors.yaml (hunk 3 above)
Comment (ericwill) [reply che-editors.yaml:56]: Which one should I keep, then?

Comment (benoitf) [reply che-editors.yaml:53]: maybe keep what is used by devworkspace now ( so secured false and protocol https) and when converting to meta.yaml, use secured true if protocol is ending with a s

**Pull request closed without merging**